frame/scheduler/src/lib.rs

```diff
@@ -376,13 +427,20 @@ impl<T: Trait> Module<T> {
 	}
 
 	fn do_schedule(
-		when: T::BlockNumber,
+		when: DelayedDispatchTime<T::BlockNumber>,
 		maybe_periodic: Option<schedule::Period<T::BlockNumber>>,
 		priority: schedule::Priority,
 		origin: T::PalletsOrigin,
 		call: <T as Trait>::Call
 	) -> Result<TaskAddress<T::BlockNumber>, DispatchError> {
-		if when <= frame_system::Module::<T>::block_number() {
+		let now = frame_system::Module::<T>::block_number();
+
+		let when = match when {
+			DelayedDispatchTime::At(x) => x,
+			DelayedDispatchTime::After(x) => now.saturating_add(x)
+		};
+
+		if when <= now {
```

Comment:
We can't raise this error in case of `DelayedDispatchTime::After`, right?

Reply:
`After(0)` is also invalid